relay-general/src/protocol/thread.rs

```diff
@@ -233,9 +233,10 @@ pub struct Thread {
     #[metastructure(skip_serialization = "empty")]
     pub state: Annotated<String>,
 
-    /// Represents an instance of a held lock (java monitor object) in a thread.
-    #[metastructure(skip_serialization = "empty")]
-    pub lock_reason: Annotated<LockReason>,
+    /// Represents a collection of locks (java monitor objects) held by a thread.
+    ///
+    /// A map of lock object addresses and their respective lock reason/details.
+    pub held_locks: Annotated<Object<LockReason>>,
```

Comment:
Consider any backward-compatibility if there already are SDKs released with `lock_reason` in the protocol.

Reply:
no, not yet, so we're good here. but thanks for the reminder!